src/client/components/pages/Schedule/SchedulePage.tsx

```diff
@@ -61,7 +77,37 @@ const SchedulePage: FunctionComponent = () => {
     selectedSemester,
     setSelectedSemester,
   ] = useStoredState<SemesterSelection>('SCHEDULE_SEMESTER_SELECTION');
+  /* Filter the courses for which data will be shown in the table
+  */
+  const [prefixes, setPrefixes] = useState <PrefixState[]>([]);
+  useEffect(() => {
+    const newPrefixes = catalogPrefixes.map((prefix) => ({
+      prefix,
+      active: true,
+    }));
+    setPrefixes(newPrefixes);
+  }, [catalogPrefixes]);
 
+  function togglePrefix(prefix:string) {
+    const prefixObj = prefixes.find((p) => p.prefix === prefix);
+    prefixObj.active = !prefixObj.active;
+    setPrefixes([...prefixes]);
+  }
+  const isPrefixActive = useCallback((prefix:string) => {
+    const prefixObj = prefixes.find((p) => p.prefix === prefix);
+    return prefixObj.active;
+  }, [prefixes]);
+  const prefixToggleButtons = prefixes.map((prefixObj) => (
+    <PrefixButton
+      alt="Course Filter Button"
+      prefix={prefixObj.prefix}
+      variant={VARIANT.BASE}
+      key={prefixObj.prefix}
+      onClick={() => togglePrefix(prefixObj.prefix)}
+    >
+      {prefixObj.prefix}
+    </PrefixButton>
+  ));
```

Comment:
I did render the `prefixToggleButtons` inside the return block by passing it in as {prefixToggleButtons} because it wouldn't let me declare it in return block. so should i declare it right before return block ?

Reply:
Right, I'm suggesting that instead of declaring a variable and passing it in, you can just call the `.map` function directly in the component, i.e.:

```tsx
return (
  <>
    <VerticalSpace>
      {prefixes.map((prefixObj) => (
        <PrefixButton
          alt="Course Filter Button"
          prefix={prefixObj.prefix}
          variant={VARIANT.BASE}
          key={prefixObj.prefix}
          onClick={() => togglePrefix(prefixObj.prefix)}
        >
          {prefixObj.prefix}
        </PrefixButton>
      ))}
...
```